database_normalization/database_normalization.md

```diff
@@ -0,0 +1,370 @@
+
+The reason why this usually doesn't happen in a database, and you instead see lots of different tables that you have to join together is because of a concept known as **normalization**.  To explain normalization, we'll first explain **one-to-many** data relationships.
+
+## One-to-Many
+
+Data often has one-to-many relationships. What does one-to-many mean? Consider two use cases, a hospital and a retail website.  Each organization has one-to-many data relationships:
+
+**Hospital one-to-many examples**
+
+* One patient has multiple encounters on different dates.
+* One encounter may include multiple orders for different procedures and medications.
+* One medication order can give rise to multiple medication administrations.
+
+**Retail website one-to-many examples**
+
+* One customer may have several associated addresses for shipping, billing, or both.
+* One customer will probably have multiple orders.
+* Each order will have one or many associated items.
+
+One-to-many data relationships (one X is related to more than one Y) can be very hard to accommodate in a single table with all the data.  Imagine having a table of order information from an online website, in which a single row contained every item included in that order. 
+
+It would be impossible to know ahead of time how many columns would be needed.  What if we provided enough columns for 100 items in an order to start?  Then, maybe someone orders 110 items.  What do we do?  Also, most orders are much smaller, which means we have many empty cells with no data in them.  It's not an efficient way to store data.  
+
+### Quiz: One-to-Many 
+
+Which of the following relationships are likely to be one-to-many?  Select all the correct answers.
+
+[[X]] Gym member to workout
+[[ ]] Street address to corresponding legislative district
+[[X]] Patient to medical provider
+[[X]] Medical provider to patient
+[[ ]] Baseball team to mascot
+[[?]] There are multiple correct answers!
+*********
+
+<div class = "answer">
+
+We hope that any gym member will have more than one recorded workout, so that relationship is one-to-many.  However, a particular street address will only have one legislative district representing residents at that address.  That's not a one-to-many relationship.  A patient can have multiple providers, so that's one-to-many, and providers can have many patients, so that's also one-to-many.  In fact you could more accurately call this relationship many-to-many!  Finally, a baseball team (we argue) should have only one mascot, so that's not a one-to-many relationship.
```

Comment:
```suggestion
We hope that any gym member will have more than one recorded workout, so that relationship is one-to-many.  However, a particular street address will only be related to a single latitude and longitude. That's not a one-to-many relationship.  A patient can have multiple providers, so that's one-to-many, and providers can have many patients, so that's also one-to-many.  In fact you could more accurately call this relationship many-to-many!  Finally, a baseball team (we argue) should have only one mascot, so that's not a one-to-many relationship.
```